[loaders] PEP 479 fix for csv loader

ajkerrigan opened the pull request:
Catch StopIteration explicitly when iterating over rows in the CSV reader.
This avoids the following error when opening CSV files in Python 3.8:

`RuntimeError: generator raised StopIteration`

but maintains graceful handling of malformed CSV files.

**References:**

https://www.python.org/dev/peps/pep-0479/#examples-of-breakage
https://github.com/python/cpython/pull/6381/files

ajkerrigan committed:
[loaders] PEP 479 fix for csv loader

Catch StopIteration explicitly when iterating over rows in the CSV reader.
This avoids the following error when opening CSV files in Python 3.8:

RuntimeError: generator raised StopIteration

But maintains the behavior of gracefully handling malformed CSV files.

References:

https://www.python.org/dev/peps/pep-0479/#examples-of-breakage
https://github.com/python/cpython/pull/6381/files

anjakefala commented:
This is the test to run before accepting! https://github.com/saulpw/visidata/blob/develop/tests/errors.vd

ajkerrigan commented:
> This is the test to run before accepting! https://github.com/saulpw/visidata/blob/develop/tests/errors.vd

Yes, thanks @anjakefala! I should have included output from that. Running that test in a 3.8 environment gave me an error before:

```
❯ ./dev/test.sh errors
--- tests/errors.vd
=== git diffs ===
Updated 0 paths from the index
===================================================
--- tests/errors.vd
opening tests/errors.vd as vd
RuntimeError: generator raised StopIteration
replay aborted during open-file
```

But with these changes, it seems clean:

```
❯ ./dev/test.sh errors
--- tests/errors.vd
=== git diffs ===
Updated 0 paths from the index
===================================================
--- tests/errors.vd
opening tests/errors.vd as vd
=== git diffs for BUILD FAILURE ===
==============================================
```

saulpw reviewed (approved):
Thanks, @ajkerrigan !  I especially appreciated the research into PEP 479.
